File: command.R
library( ANTsR )

# Read in the image

dataDirectory <- './Input/'

inputImage <- antsImageRead(
   paste0( dataDirectory, 't1_slice.nii.gz' ), dimension = 2 )

outputImage <- denoiseImage( inputImage, verbose = TRUE )

outputDirectory <- './OutputANTsR/'
if( ! dir.exists( outputDirectory ) )
  {
  dir.create( outputDirectory )
  }
antsImageWrite( outputImage, paste0( outputDirectory, "t1_slice_denoised.nii.gz" ) )

File: command.py
import ants
import os

# os.chdir( "/Users/ntustison/Data/antsExamples/DenoiseImageExample/" )

# Read in the image

dataDirectory = './Input/'

inputImageFile = dataDirectory + 't1_slice.nii.gz'
inputImage = ants.image_read( inputImageFile )

outputImage = ants.denoise_image( inputImage, v = 1 )

outputDirectory = './OutputANTsPy/'
if not os.path.isdir( outputDirectory ):
  os.mkdir( outputDirectory )

outputImageFile = outputDirectory + "t1_slice_denoised.nii.gz"
ants.image_write( outputImage, outputImageFile )

